armi/reactor/components/component.py

```diff
@@ -1262,6 +1264,72 @@ def getIntegratedMgFlux(self, adjoint=False, gamma=False):
 
         return pinFluxes[self.p.pinNum - 1] * self.getVolume()
 
+    def getPinMgFluxes(
+        self, adjoint: Optional[bool] = False, gamma: Optional[bool] = False
+    ) -> np.ndarray:
+        """Retrieves the pin multigroup fluxes for the component.
+
+        Parameters
+        ----------
+        adjoint : bool, optional
+            Return adjoint flux instead of real
+        gamma : bool, optional
+            Whether to return the neutron flux or the gamma flux.
+
+        Returns
+        -------
+        np.ndarray
+            A ``(N, nGroup)`` array of pin multigroup fluxes, where ``N`` is the
+            equivalent to the multiplicity of the component (``self.p.mult``)
+            and ``nGroup`` is the number of energy groups of the flux.
+
+        Raises
+        ------
+        ValueError
+            If the location(s) of the component are not aligned with pin indices
+            from the block. This would happen if this component is not actually
+            a pin.
+        """
+        # Get the (i, j, k) location of all pins from the parent block
+        # FIXME: This should be changed to just using Block.getPinLocations once Drew's PR is merged
+        indicesAll = []
+        for clad in self.parent.getChildrenWithFlags(flags.Flags.CLAD):
+            if isinstance(clad.spatialLocator, grids.MultiIndexLocation):
+                indicesAll.extend(clad.spatialLocator.indices)
+            else:
+                indicesAll.append(clad.spatialLocator.indices)
+
+        # Retrieve the indices of this component
+        indices = self.spatialLocator.indices
+        if not isinstance(indices, list):
+            indices = [indices]
+
+        # Map this component's indices to block's pin indices
+        getIndex = lambda ind: next(
+            (i for i, indA in enumerate(indicesAll) if np.all(ind == indA)),
+            None,
+        )
+        indexMap = list(map(getIndex, indices))
+        if None in indexMap:
+            msg = f"Failed to retrieve pin indices for component {self}."
+            runLog.error(msg)
+            raise ValueError(msg)
+
+        # Get the parameter name we are trying to retrieve
+        if gamma:
+            if adjoint:
+                raise ValueError("Adjoint gamma flux is currently unsupported.")
+            else:
+                param = "pinMgFluxesGamma"
+        else:
+            if adjoint:
+                param = "pinMgFluxesAdj"
+            else:
+                param = "pinMgFluxes"
+
+        # Return pin fluxes
+        return self.parent.p.get(param)[indexMap]
```

Comment:
If the parameter `param` is not defined on the parent, or is `None`, there could be a not great exception raised here. I would recommend some error helping, maybe a try/except/raise pattern like
```python
try:
    return self.parent.p.get(param)[indexMap]
except Exception as ee:
    msg = f"Failure getting {param} from {self} via parent {self.parent}"
    runLog.error(msg)
    runLog.error(ee)
    raise ValueError(msg) from ee
```
or just allowing the raised exception to pass up, replacing the `raise ValueError(msg) from ee` with `raise`